modules/javafx.base/src/main/java/javafx/beans/property/IntegerPropertyBase.java

```diff
@@ -53,7 +53,7 @@
 public abstract class IntegerPropertyBase extends IntegerProperty {
 
     private int value;
-    private ObservableIntegerValue observable = null;;
```

Comment:
Yes, `potential programming problems / empty statement`. It can sometimes point to a mistake where a semi colon is used just before a block.

Reply:
Cool.  My eclipse finds other places which do look like a bug - see disposed:65
It is in the base - why isn't it this PR?

Reply:
full list

```
Description	Resource	Location
Empty control-flow statement	DateCellBehavior.java	line 95
Empty control-flow statement	Disposer.java	line 65
Empty control-flow statement	Disposer.java	line 66
Empty control-flow statement	Disposer.java	line 66
Empty control-flow statement	Disposer.java	line 72
Empty control-flow statement	DualPivotQuicksort20191112Ext.java	line 267
Empty control-flow statement	DualPivotQuicksort20191112Ext.java	line 268
Empty control-flow statement	DualPivotQuicksort20191112Ext.java	line 380
Empty control-flow statement	DualPivotQuicksort20191112Ext.java	line 494
Empty control-flow statement	DualPivotQuicksort20191112Ext.java	line 673
Empty control-flow statement	DualPivotQuicksort20191112Ext.java	line 678
Empty control-flow statement	DualPivotQuicksort20191112Ext.java	line 686
Empty control-flow statement	DualPivotQuicksort20191112Ext.java	line 800
Empty control-flow statement	DualPivotQuicksort20191112Ext.java	line 808
Empty control-flow statement	KeyboardShortcutsHandler.java	line 498
Empty control-flow statement	PrismFontFactory.java	line 1401
Unnecessary semicolon	AbstractPrimaryTimerTest.java	line 188
Unnecessary semicolon	Activity.java	line 31
Unnecessary semicolon	AnimationMock.java	line 41
Unnecessary semicolon	Arc2D.java	line 388
Unnecessary semicolon	BaseMesh.java	line 68
Unnecessary semicolon	BindingsNumberCalculationsTest.java	line 838
Unnecessary semicolon	Bloom.java	line 90
Unnecessary semicolon	BoxBlur.java	line 95
Unnecessary semicolon	ButtonBaseTest.java	line 182
Unnecessary semicolon	CachingShapeRep.java	line 234
Unnecessary semicolon	CascadingStyle.java	line 184
Unnecessary semicolon	CellUtils.java	line 155
Unnecessary semicolon	CellUtils.java	line 305
Unnecessary semicolon	ChoiceBoxTreeCell.java	line 340
Unnecessary semicolon	Clipboard.java	line 389
Unnecessary semicolon	ClipboardContent.java	line 212
Unnecessary semicolon	ColorAdjust.java	line 98
Unnecessary semicolon	ColorInput.java	line 83
Unnecessary semicolon	ColorPalette.java	line 445
Unnecessary semicolon	ComboBoxTreeCell.java	line 373
Unnecessary semicolon	ContextMenuTest.java	line 259
Unnecessary semicolon	CssParser.java	line 2493
Unnecessary semicolon	CssParser.java	line 3830
Unnecessary semicolon	CssParser.java	line 3834
Unnecessary semicolon	D3DFrameStats.java	line 54
Unnecessary semicolon	DefaultCancelButtonTestBase.java	line 304
Unnecessary semicolon	DisplacementMap.java	line 108
Unnecessary semicolon	DisplacementMap.java	line 243
Unnecessary semicolon	DropShadow.java	line 148
Unnecessary semicolon	DummyTexture.java	line 41
Unnecessary semicolon	DumpRenderTree.java	line 94
Unnecessary semicolon	DumpRenderTree.java	line 106
Unnecessary semicolon	FXCanvas.java	line 171
Unnecessary semicolon	FXCanvas.java	line 1254
Unnecessary semicolon	FXDnDInteropN.java	line 374
Unnecessary semicolon	FireButtonBaseTest.java	line 92
Unnecessary semicolon	GaussianBlur.java	line 83
Unnecessary semicolon	Glow.java	line 82
Unnecessary semicolon	HyperlinkTest.java	line 250
Unnecessary semicolon	Image.java	line 248
Unnecessary semicolon	ImageInput.java	line 81
Unnecessary semicolon	ImageStorage.java	line 115
Unnecessary semicolon	InnerShadow.java	line 128
Unnecessary semicolon	IosImageLoaderFactory.java	line 42
Unnecessary semicolon	J2DPrinter.java	line 865
Unnecessary semicolon	J2DPrinterJob.java	line 776
Unnecessary semicolon	KeyCode.java	line 1177
Unnecessary semicolon	KeyCodeMap.java	line 62
Unnecessary semicolon	KeyFrameTest.java	line 131
Unnecessary semicolon	KeyFrameTest.java	line 136
Unnecessary semicolon	KeyFrameTest.java	line 141
Unnecessary semicolon	KeyFrameTest.java	line 146
Unnecessary semicolon	KeyFrameTest.java	line 218
Unnecessary semicolon	KeyFrameTest.java	line 223
Unnecessary semicolon	KeyFrameTest.java	line 228
Unnecessary semicolon	LeakTest.java	line 187
Unnecessary semicolon	LeakTest.java	line 348
Unnecessary semicolon	LeakTest.java	line 474
Unnecessary semicolon	LeakTest.java	line 205
Unnecessary semicolon	LeakTest.java	line 210
Unnecessary semicolon	LeakTest.java	line 215
Unnecessary semicolon	LightBaseTest.java	line 89
Unnecessary semicolon	Lighting.java	line 77
Unnecessary semicolon	Lighting.java	line 164
Unnecessary semicolon	LineChartTest.java	line 55
Unnecessary semicolon	LinearConvolveRenderState.java	line 124
Unnecessary semicolon	LinuxArch.java	line 37
Unnecessary semicolon	ListCellTest.java	line 918
Unnecessary semicolon	ListView.java	line 526
Unnecessary semicolon	ListViewTest.java	line 775
Unnecessary semicolon	ListViewTest.java	line 815
Unnecessary semicolon	MacAccessible.java	line 1122
Unnecessary semicolon	MacTouchInputSupport.java	line 63
Unnecessary semicolon	MediaException.java	line 89
Unnecessary semicolon	MediaException.java	line 201
Unnecessary semicolon	MediaPlayer.java	line 267
Unnecessary semicolon	MediaView.java	line 475
Unnecessary semicolon	MediaView.java	line 744
Unnecessary semicolon	MediaView.java	line 800
Unnecessary semicolon	MenuItemTest.java	line 351
Unnecessary semicolon	MenuItemTest.java	line 541
Unnecessary semicolon	MenuTest.java	line 598
Unnecessary semicolon	MotionBlur.java	line 87
Unnecessary semicolon	NodeEffectInput.java	line 48
Unnecessary semicolon	PaintTypeTest.java	line 74
Unnecessary semicolon	ParsedValueTest.java	line 64
Unnecessary semicolon	PerspectiveTransform.java	line 136
Unnecessary semicolon	PhongMaterial.java	line 35
Unnecessary semicolon	PlayerStateEvent.java	line 37
Unnecessary semicolon	PopupWindow.java	line 692
Unnecessary semicolon	PredefinedMeshManager.java	line 234
Unnecessary semicolon	Printer.java	line 215
Unnecessary semicolon	PrinterJob.java	line 473
Unnecessary semicolon	PrismTextLayout.java	line 983
Unnecessary semicolon	PrismTrace.java	line 56
Unnecessary semicolon	PseudoClassTest.java	line 783
Unnecessary semicolon	PseudoClassTest.java	line 808
Unnecessary semicolon	PseudoClassTest.java	line 833
Unnecessary semicolon	PseudoClassTest.java	line 861
Unnecessary semicolon	PseudoClassTest.java	line 909
Unnecessary semicolon	PseudoClassTest.java	line 930
Unnecessary semicolon	PseudoClassTest.java	line 945
Unnecessary semicolon	PseudoClassTest.java	line 966
Unnecessary semicolon	PseudoClassTest.java	line 981
Unnecessary semicolon	PseudoClassTest.java	line 1002
Unnecessary semicolon	PseudoClassTest.java	line 1045
Unnecessary semicolon	PseudoClassTest.java	line 1050
Unnecessary semicolon	PseudoClassTest.java	line 1099
Unnecessary semicolon	PseudoClassTest.java	line 1104
Unnecessary semicolon	PseudoClassTest.java	line 1141
Unnecessary semicolon	PseudoClassTest.java	line 1199
Unnecessary semicolon	PseudoClassTest.java	line 1221
Unnecessary semicolon	PseudoClassTest.java	line 1242
Unnecessary semicolon	PseudoClassTest.java	line 1246
Unnecessary semicolon	PseudoClassTest.java	line 1265
Unnecessary semicolon	Reflection.java	line 95
Unnecessary semicolon	Region.java	line 978
Unnecessary semicolon	Region.java	line 1426
Unnecessary semicolon	RegionTest.java	line 1224
Unnecessary semicolon	RenderThemeImpl.java	line 84
Unnecessary semicolon	RenderThemeImpl.java	line 445
Unnecessary semicolon	RenderThemeImpl.java	line 463
Unnecessary semicolon	RenderThemeImpl.java	line 477
Unnecessary semicolon	RenderThemeImpl.java	line 491
Unnecessary semicolon	RenderThemeImpl.java	line 504
Unnecessary semicolon	RenderThemeImpl.java	line 522
Unnecessary semicolon	RenderThemeImpl.java	line 542
Unnecessary semicolon	RenderThemeImpl.java	line 592
Unnecessary semicolon	Scene.java	line 2018
Unnecessary semicolon	SepiaTone.java	line 83
Unnecessary semicolon	Shadow.java	line 92
Unnecessary semicolon	SplitPane.java	line 260
Unnecessary semicolon	StyleManagerTest.java	line 1007
Unnecessary semicolon	StyleManagerTest.java	line 1017
Unnecessary semicolon	SubScene.java	line 143
Unnecessary semicolon	SwingNodeHelper.java	line 117
Unnecessary semicolon	TableColumnHeaderTest.java	line 291
Unnecessary semicolon	TableHeaderRow.java	line 264
Unnecessary semicolon	TableViewTest.java	line 1535
Unnecessary semicolon	TableViewTest.java	line 1587
Unnecessary semicolon	TableViewTest.java	line 1743
Unnecessary semicolon	TableViewTest.java	line 1785
Unnecessary semicolon	TextArea.java	line 432
Unnecessary semicolon	TextAreaSkin.java	line 496
Unnecessary semicolon	TextBehaviorShim.java	line 46
Unnecessary semicolon	TextInputControlBehavior.java	line 642
Unnecessary semicolon	TextInputControlSkin.java	line 125
Unnecessary semicolon	TextInputControlSkin.java	line 145
Unnecessary semicolon	TextInputControlSkin.java	line 740
Unnecessary semicolon	TextInputControlSkin.java	line 745
Unnecessary semicolon	TextInputControlSkin.java	line 750
Unnecessary semicolon	TextInputControlSkin.java	line 838
Unnecessary semicolon	ToolBar.java	line 152
Unnecessary semicolon	Tooltip.java	line 218
Unnecessary semicolon	TreeCellTest.java	line 1001
Unnecessary semicolon	TreeTableColumnHeaderTest.java	line 120
Unnecessary semicolon	TreeTableViewTest.java	line 2522
Unnecessary semicolon	TreeTableViewTest.java	line 2575
Unnecessary semicolon	TreeTableViewTest.java	line 2718
Unnecessary semicolon	TreeTableViewTest.java	line 2757
Unnecessary semicolon	TreeViewTest.java	line 1121
Unnecessary semicolon	TreeViewTest.java	line 1160
Unnecessary semicolon	URLTypeTest.java	line 96
Unnecessary semicolon	VirtualizedControlTestUtils.java	line 95
Unnecessary semicolon	WCPasteboard.java	line 41
Unnecessary semicolon	WCPluginWidget.java	line 52
Unnecessary semicolon	WinHTMLCodec.java	line 169
Unnecessary semicolon	WinHTMLCodec.java	line 170
Unnecessary semicolon	WinHTMLCodec.java	line 384
Unnecessary semicolon	XYChart.java	line 1329

```

Reply:
many of those ^^ appear to be bugs.

Reply:
I didn't fix all problems, only the easy non-controversial ones.  Things that might actually be bugs require more careful review and so are best saved for smaller PR's IMHO.

This is already fixing around 2000 warnings in 250 files, and before going in even deeper I'd like to see if this will actually get accepted into JavaFX in a reasonable time frame (PR's changing large numbers of files don't age well).

Reply:
I only fixed the warnings listed in the ticket :)

Reply:
but the ticket specifies (javafx.)base, right?

Reply:
let's create a followup ticket - the messed up statements in Disposer and especially in PrismFontFactory:1401 are worrying.

cc: @prrace

Reply:
> but the ticket specifies (javafx.)base, right?

Yes, so only the warnings listed and only in javafx.base.  Rawtype/unchecked warnings, and things like empty control-flow statements, unused variables/fields/types and dead code will be fixed in other tickets.  It's possible some of these won't be fixed if it is an unusual but harmless pattern. Generally it's better to fix or slightly rewrite these though as it makes things easier when there are no exceptions (ie. when enforcing things later).

Reply:
@andy-goryachev-oracle The ones in the various `Disposer` classes and `PrismFontFactory` are functionally correct, but use an anti-pattern. The block really is intentionally empty, but the better pattern for this is to use `{}` rather than a `;` since the latter looks like you might have made a mistake.

Reply:
you are right - these are not bugs.